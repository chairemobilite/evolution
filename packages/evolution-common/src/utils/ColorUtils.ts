/*
 * Copyright 2025, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */

/**
 * Converts a hex color string to an RGB array
 * @param hex - The hex color string (example: #000000)
 * @returns The RGB array or undefined if the hex color string is invalid (example: rgb(0, 0, 0))
 */
export const hexToRgb = (hex: string): [number, number, number] | undefined => {
    if (!hex.startsWith('#')) {
        // if the hex color string is not prefixed with a #, add it
        hex = `#${hex}`;
    }
    const cleanHex = hex.replace('#', '').toLowerCase();
    const match = cleanHex.match(/[0-9a-f]{2}/g);
    if (match && match.length === 3) {
        return [parseInt(match[0], 16), parseInt(match[1], 16), parseInt(match[2], 16)];
    } else {
        console.error(`Invalid hex color string: ${hex}`);
        return undefined;
    }
};
