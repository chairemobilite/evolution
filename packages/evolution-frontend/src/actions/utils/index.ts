export * from './Conditional';
export * from './Validation';
export * from './WidgetOperation';
