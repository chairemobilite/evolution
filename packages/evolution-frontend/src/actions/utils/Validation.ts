import { UserInterviewAttributes } from 'evolution-common/lib/services/interviews/interview';
import { ValidationFunction } from 'evolution-common/lib/services/widgets';
import { I18nData } from 'evolution-common/lib/utils/helpers';

/**
 * Check the validations for a specific value
 * @param validations The validation functions or undefined if no function
 * @param value The value of the widget
 * @param customValue Any custom value associated with this widget
 * @param interview The interview data
 * @param path The path of the current widget to validate
 * @param customPath The custom path
 * @returns An array where the first value is a boolean indicating if the value
 * is valid and the second value is the error message if the value is invalid
 */
export const checkValidations = <CustomSurvey, CustomHousehold, CustomHome, CustomPerson>(
    validations: ValidationFunction<CustomSurvey, CustomHousehold, CustomHome, CustomPerson> | undefined,
    value: unknown | undefined,
    customValue: unknown | undefined,
    interview: UserInterviewAttributes<CustomSurvey, CustomHousehold, CustomHome, CustomPerson>,
    path: string,
    customPath?: string
): [boolean, I18nData<CustomSurvey, CustomHousehold, CustomHome, CustomPerson> | undefined] => {
    if (typeof validations === 'function') {
        try {
            const validationsGroup = validations(value, customValue, interview, path, customPath);
            for (let i = 0; i < validationsGroup.length; i++) {
                if (validationsGroup[i].validation === true) {
                    return [false, validationsGroup[i].errorMessage];
                }
            }
        } catch (error) {
            // If there is an error during validations, just ignore the error and consider the value as valid to make sure we can at least get the next answers without blocking the questionnaire.
            // TODO: add a server-side log of the error and define a better way to deal with errors.
            console.log('validation error', error);
            return [true, undefined];
        }
    }
    return [true, undefined];
};

export const validateAllWidgets = function <CustomSurvey, CustomHousehold, CustomHome, CustomPerson>(
    interview: UserInterviewAttributes<CustomSurvey, CustomHousehold, CustomHome, CustomPerson>
) {
    // TODO Move the allWidgetsValid out of the interview type
    return (interview as any).allWidgetsValid;
};
