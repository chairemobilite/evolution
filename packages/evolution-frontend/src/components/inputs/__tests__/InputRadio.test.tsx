/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { faCrow } from '@fortawesome/free-solid-svg-icons/faCrow';

import { interviewAttributes } from './interviewData.test';

import { shuffle } from 'chaire-lib-common/lib/utils/RandomUtils';

import InputRadio from '../InputRadio';
import i18next from 'i18next';

jest.mock('chaire-lib-common/lib/utils/RandomUtils', () => ({
    shuffle: jest.fn()
}));

const shuffleMock = shuffle as jest.MockedFunction<typeof shuffle>;

const userAttributes = {
    id: 1,
    username: 'foo',
    preferences: {  },
    serializedPermissions: [],
    isAuthorized: () => true,
    is_admin: false,
    pages: [],
    showUserInfo: true
}

describe('Render InputRadio with various parameter combinations, all parameters', () => {

    const conditionalFct = jest.fn().mockReturnValue(true);
    const translationFct = jest.fn().mockReturnValue('Translated string');
    const choices = [
        {
            value: 'val1',
            label: { en: 'english value', fr: 'valeur française' },
            hidden: false,
            icon: faCrow
        },
        {
            value: 'val2',
            label: 'Unilingual label',
            iconPath: 'img/test.png'
        },
        {
            value: 'hiddenVal',
            label: { en: 'english hidden', fr: 'cachée' },
            hidden: true
        },
        {
            value: 'conditionalVal',
            label: { en: 'english conditional', fr: 'conditionnelle' },
            conditional: conditionalFct
        },
        {
            value: 'val3',
            label: translationFct,
            hidden: false,
            color: 'green'
        }
    ];

    const widgetConfig = {
        type: 'question' as const,
        twoColumns: true,
        path: 'test.foo',
        inputType: 'radio' as const,
        choices,
        iconSize: '3em',
        seed: 23,
        shuffle: false,
        addCustom: false,
        columns: 2,
        size: 'medium' as const,
        sameLine: false,
        customLabel: 'custom label',
        containsHtml: true,
        label: {
            fr: `Texte en français`,
            en: `English text`
        }
    };

    beforeEach(() => {
        conditionalFct.mockClear();
    })

    test('Includes hidden values, conditional displayed, no custom, 2 columns', () => {
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */}}
                widgetConfig={widgetConfig}
                value='value'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
            />
        );
        expect(wrapper).toMatchSnapshot();
        expect(conditionalFct).toHaveBeenCalledTimes(1);
        expect(conditionalFct).toHaveBeenCalledWith(interviewAttributes, 'foo.test', userAttributes);
        expect(translationFct).toHaveBeenCalledWith(i18next.t, interviewAttributes, 'foo.test', userAttributes);
    });

    test('Conditional value hidden, same line', () => {
        conditionalFct.mockReturnValueOnce(false);
        const testWidgetConfig = Object.assign({}, widgetConfig, { sameLine: true, columns: undefined });
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */}}
                widgetConfig={testWidgetConfig}
                value='value'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('Shuffle choices', () => {
        shuffleMock.mockReturnValue([choices[1], choices[3], choices[0], choices[2]]);
        const testWidgetConfig = Object.assign({}, widgetConfig, { shuffle: true });
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */}}
                widgetConfig={testWidgetConfig}
                value='value'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
            />
        );
        expect(wrapper).toMatchSnapshot();
        expect(shuffleMock).toHaveBeenCalledTimes(1);
        expect(shuffleMock).toHaveBeenCalledWith(choices, undefined, widgetConfig.seed);
    });

    test('With custom choice', () => {
        const testWidgetConfig = Object.assign({}, widgetConfig, { addCustom: true, customChoice: 'val2' });
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */ }}
                widgetConfig={testWidgetConfig}
                value='val2'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
                customId='foo.test.custom'
                
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    
});

describe('Render InputRadio with minimum parameters', () => {

    const choices = [
        {
            value: 'val1',
            label: { en: 'english value', fr: 'valeur française' }
        },
        {
            value: 'val2',
            label: 'Unilingual label'
        }
    ];

    const widgetConfig = {
        type: 'question' as const,
        twoColumns: true,
        path: 'test.foo',
        inputType: 'radio' as const,
        choices,
        containsHtml: true,
        label: {
            fr: `Texte en français`,
            en: `English text`
        }
    };

    test('Minimum parameters', () => {
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */}}
                widgetConfig={widgetConfig}
                value='value'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe('Render InputRadio with HTML label', () => {

    const choices = [
        {
            value: 'first value',
            label: {
                fr: '<div>premiere valeur</div>',
                en: '<div>first value</div>'
            }
        },
        {
            value: 'second value',
            label: '<div>second value</div>'
        }
    ];

    const widgetConfig = {
        type: 'question' as const,
        twoColumns: true,
        path: 'test.foo',
        inputType: 'radio' as const,
        choices,
        size: 'medium',
        containsHtml: true,
        label: {
            fr: `Texte en français`,
            en: `English text`
        }
    };

    test('HTML label', () => {
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */}}
                widgetConfig={widgetConfig}
                value='value'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});

describe('Render InputRadio with various alignments', () =>{

    const choices = [
        {
            value: 'first value',
            label: {
                fr: '<div>premiere valeur</div>',
                en: '<div>first value</div>'
            }
        },
        {
            value: 'second value',
            label: '<div>second value</div>'
        }
    ];

    const widgetConfig = {
        type: 'question' as const,
        twoColumns: true,
        path: 'test.foo',
        inputType: 'radio' as const,
        alignment: undefined,
        choices,
        containsHtml: true,
        label: {
            fr: `Texte en français`,
            en: `English text`
        }
    };

    test('Undefined alignment with no rows and no columns', () =>{
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */}}
                widgetConfig={widgetConfig}
                value='value'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
            />
        );
        expect(wrapper).toMatchSnapshot();
    })
    test('Auto alignment with 2 columns', () =>{
        const testWidgetConfig = Object.assign({}, widgetConfig, { alignment: 'auto', columns: 2 });
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */}}
                widgetConfig={testWidgetConfig}
                value='value'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
            />
        );
        expect(wrapper).toMatchSnapshot();
    })
    test('Vertical alignment with 4 rows', () =>{
        const testWidgetConfig = Object.assign({}, widgetConfig, { alignment: 'vertical', customAlignmentLengths: [1,1] });
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */}}
                widgetConfig={testWidgetConfig}
                value='value'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
            />
        );
        expect(wrapper).toMatchSnapshot();
    })
    test('Horizontal alignment with custom alignments', () =>{
        const testWidgetConfig = Object.assign({}, widgetConfig, { alignment: 'horizontal', rows: 4 });
        const wrapper = TestRenderer.create(
            <InputRadio
                id={'test'}
                onValueChange={() => { /* nothing to do */}}
                widgetConfig={testWidgetConfig}
                value='value'
                inputRef={React.createRef()}
                interview={interviewAttributes}
                user={userAttributes}
                path='foo.test'
            />
        );
        expect(wrapper).toMatchSnapshot();
    })
});
