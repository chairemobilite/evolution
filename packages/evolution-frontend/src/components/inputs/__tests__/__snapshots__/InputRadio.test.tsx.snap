// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render InputRadio with HTML label HTML label 1`] = `
<div
  className="survey-question__input-radio-group-container"
>
  <div
    className="survey-question__input-radio-input-container"
    onClick={[Function]}
  >
    <div
      className="label-input-container"
    >
      <input
        checked={false}
        className="input-radio"
        id="test_first value__input-radio__first value"
        name="test_first value"
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value="first value"
      />
      <label
        htmlFor="test_first value__input-radio__first value"
        onClick={[Function]}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<div>first value</div>",
            }
          }
        />
      </label>
    </div>
  </div>
  <div
    className="survey-question__input-radio-input-container"
    onClick={[Function]}
  >
    <div
      className="label-input-container"
    >
      <input
        checked={false}
        className="input-radio"
        id="test_second value__input-radio__second value"
        name="test_second value"
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value="second value"
      />
      <label
        htmlFor="test_second value__input-radio__second value"
        onClick={[Function]}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<div>second value</div>",
            }
          }
        />
      </label>
    </div>
  </div>
</div>
`;

exports[`Render InputRadio with minimum parameters Minimum parameters 1`] = `
<div
  className="survey-question__input-radio-group-container"
>
  <div
    className="survey-question__input-radio-input-container"
    onClick={[Function]}
  >
    <div
      className="label-input-container"
    >
      <input
        checked={false}
        className="input-radio"
        id="test_val1__input-radio__val1"
        name="test_val1"
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value="val1"
      />
      <label
        htmlFor="test_val1__input-radio__val1"
        onClick={[Function]}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "english value",
            }
          }
        />
      </label>
    </div>
  </div>
  <div
    className="survey-question__input-radio-input-container"
    onClick={[Function]}
  >
    <div
      className="label-input-container"
    >
      <input
        checked={false}
        className="input-radio"
        id="test_val2__input-radio__val2"
        name="test_val2"
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value="val2"
      />
      <label
        htmlFor="test_val2__input-radio__val2"
        onClick={[Function]}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Unilingual label",
            }
          }
        />
      </label>
    </div>
  </div>
</div>
`;

exports[`Render InputRadio with various parameter combinations, all parameters Conditional value hidden, same line 1`] = `
<div
  className="survey-question__input-radio-group-container"
>
  <div
    className="survey-question__input-radio-input-container"
    onClick={[Function]}
  >
    <div
      className="label-input-container"
    >
      <input
        checked={false}
        className="input-radio"
        id="test_val1__input-radio__val1"
        name="test_val1"
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value="val1"
      />
      <label
        htmlFor="test_val1__input-radio__val1"
        onClick={[Function]}
      >
        <span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-crow fa-w-20 faIconLeft"
            data-icon="crow"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={
              Object {
                "margin": "auto 0",
              }
            }
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M544 32h-16.36C513.04 12.68 490.09 0 464 0c-44.18 0-80 35.82-80 80v20.98L12.09 393.57A30.216 30.216 0 0 0 0 417.74c0 22.46 23.64 37.07 43.73 27.03L165.27 384h96.49l44.41 120.1c2.27 6.23 9.15 9.44 15.38 7.17l22.55-8.21c6.23-2.27 9.44-9.15 7.17-15.38L312.94 384H352c1.91 0 3.76-.23 5.66-.29l44.51 120.38c2.27 6.23 9.15 9.44 15.38 7.17l22.55-8.21c6.23-2.27 9.44-9.15 7.17-15.38l-41.24-111.53C485.74 352.8 544 279.26 544 192v-80l96-16c0-35.35-42.98-64-96-64zm-80 72c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "english value",
            }
          }
        />
      </label>
    </div>
  </div>
  <div
    className="survey-question__input-radio-input-container"
    onClick={[Function]}
  >
    <div
      className="label-input-container"
    >
      <input
        checked={false}
        className="input-radio"
        id="test_val2__input-radio__val2"
        name="test_val2"
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value="val2"
      />
      <label
        htmlFor="test_val2__input-radio__val2"
        onClick={[Function]}
      >
        <span>
          <img
            alt="Unilingual label"
            className="faIconLeft"
            src="img/test.png"
            style={
              Object {
                "height": "3em",
                "margin": "0 0.25em 0 0",
              }
            }
          />
        </span>
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Unilingual label",
            }
          }
        />
      </label>
    </div>
  </div>
  <div
    className="survey-question__input-radio-input-container"
    onClick={[Function]}
  >
    <div
      className="label-input-container"
    >
      <input
        checked={false}
        className="input-radio"
        id="test_val3__input-radio__val3"
        name="test_val3"
        onChange={[Function]}
        onClick={[Function]}
        type="radio"
        value="val3"
      />
      <label
        htmlFor="test_val3__input-radio__val3"
        onClick={[Function]}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Translated string",
            }
          }
        />
      </label>
    </div>
  </div>
</div>
`;

exports[`Render InputRadio with various parameter combinations, all parameters Includes hidden values, conditional displayed, no custom, 2 columns 1`] = `
<div
  className="survey-question__input-radio-group-container no-wrap"
>
  <div
    className="survey-question__input-radio-group-column"
  >
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_val1__input-radio__val1"
          name="test_val1"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="val1"
        />
        <label
          htmlFor="test_val1__input-radio__val1"
          onClick={[Function]}
        >
          <span>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-crow fa-w-20 faIconLeft"
              data-icon="crow"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "margin": "auto 0",
                }
              }
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M544 32h-16.36C513.04 12.68 490.09 0 464 0c-44.18 0-80 35.82-80 80v20.98L12.09 393.57A30.216 30.216 0 0 0 0 417.74c0 22.46 23.64 37.07 43.73 27.03L165.27 384h96.49l44.41 120.1c2.27 6.23 9.15 9.44 15.38 7.17l22.55-8.21c6.23-2.27 9.44-9.15 7.17-15.38L312.94 384H352c1.91 0 3.76-.23 5.66-.29l44.51 120.38c2.27 6.23 9.15 9.44 15.38 7.17l22.55-8.21c6.23-2.27 9.44-9.15 7.17-15.38l-41.24-111.53C485.74 352.8 544 279.26 544 192v-80l96-16c0-35.35-42.98-64-96-64zm-80 72c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "english value",
              }
            }
          />
        </label>
      </div>
    </div>
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_val2__input-radio__val2"
          name="test_val2"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="val2"
        />
        <label
          htmlFor="test_val2__input-radio__val2"
          onClick={[Function]}
        >
          <span>
            <img
              alt="Unilingual label"
              className="faIconLeft"
              src="img/test.png"
              style={
                Object {
                  "height": "3em",
                  "margin": "0 0.25em 0 0",
                }
              }
            />
          </span>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Unilingual label",
              }
            }
          />
        </label>
      </div>
    </div>
  </div>
  <div
    className="survey-question__input-radio-group-column"
  >
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_conditionalVal__input-radio__conditionalVal"
          name="test_conditionalVal"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="conditionalVal"
        />
        <label
          htmlFor="test_conditionalVal__input-radio__conditionalVal"
          onClick={[Function]}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "english conditional",
              }
            }
          />
        </label>
      </div>
    </div>
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_val3__input-radio__val3"
          name="test_val3"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="val3"
        />
        <label
          htmlFor="test_val3__input-radio__val3"
          onClick={[Function]}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Translated string",
              }
            }
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render InputRadio with various parameter combinations, all parameters Shuffle choices 1`] = `
<div
  className="survey-question__input-radio-group-container no-wrap"
>
  <div
    className="survey-question__input-radio-group-column"
  >
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_val2__input-radio__val2"
          name="test_val2"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="val2"
        />
        <label
          htmlFor="test_val2__input-radio__val2"
          onClick={[Function]}
        >
          <span>
            <img
              alt="Unilingual label"
              className="faIconLeft"
              src="img/test.png"
              style={
                Object {
                  "height": "3em",
                  "margin": "0 0.25em 0 0",
                }
              }
            />
          </span>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Unilingual label",
              }
            }
          />
        </label>
      </div>
    </div>
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_conditionalVal__input-radio__conditionalVal"
          name="test_conditionalVal"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="conditionalVal"
        />
        <label
          htmlFor="test_conditionalVal__input-radio__conditionalVal"
          onClick={[Function]}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "english conditional",
              }
            }
          />
        </label>
      </div>
    </div>
  </div>
  <div
    className="survey-question__input-radio-group-column"
  >
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_val1__input-radio__val1"
          name="test_val1"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="val1"
        />
        <label
          htmlFor="test_val1__input-radio__val1"
          onClick={[Function]}
        >
          <span>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-crow fa-w-20 faIconLeft"
              data-icon="crow"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "margin": "auto 0",
                }
              }
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M544 32h-16.36C513.04 12.68 490.09 0 464 0c-44.18 0-80 35.82-80 80v20.98L12.09 393.57A30.216 30.216 0 0 0 0 417.74c0 22.46 23.64 37.07 43.73 27.03L165.27 384h96.49l44.41 120.1c2.27 6.23 9.15 9.44 15.38 7.17l22.55-8.21c6.23-2.27 9.44-9.15 7.17-15.38L312.94 384H352c1.91 0 3.76-.23 5.66-.29l44.51 120.38c2.27 6.23 9.15 9.44 15.38 7.17l22.55-8.21c6.23-2.27 9.44-9.15 7.17-15.38l-41.24-111.53C485.74 352.8 544 279.26 544 192v-80l96-16c0-35.35-42.98-64-96-64zm-80 72c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "english value",
              }
            }
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render InputRadio with various parameter combinations, all parameters With custom choice 1`] = `
<div
  className="survey-question__input-radio-group-container no-wrap"
>
  <div
    className="survey-question__input-radio-group-column"
  >
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_val1__input-radio__val1"
          name="test_val1"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="val1"
        />
        <label
          htmlFor="test_val1__input-radio__val1"
          onClick={[Function]}
        >
          <span>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-crow fa-w-20 faIconLeft"
              data-icon="crow"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={
                Object {
                  "margin": "auto 0",
                }
              }
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M544 32h-16.36C513.04 12.68 490.09 0 464 0c-44.18 0-80 35.82-80 80v20.98L12.09 393.57A30.216 30.216 0 0 0 0 417.74c0 22.46 23.64 37.07 43.73 27.03L165.27 384h96.49l44.41 120.1c2.27 6.23 9.15 9.44 15.38 7.17l22.55-8.21c6.23-2.27 9.44-9.15 7.17-15.38L312.94 384H352c1.91 0 3.76-.23 5.66-.29l44.51 120.38c2.27 6.23 9.15 9.44 15.38 7.17l22.55-8.21c6.23-2.27 9.44-9.15 7.17-15.38l-41.24-111.53C485.74 352.8 544 279.26 544 192v-80l96-16c0-35.35-42.98-64-96-64zm-80 72c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "english value",
              }
            }
          />
        </label>
      </div>
    </div>
    <div
      className="survey-question__input-radio-input-container checked"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={true}
          className="input-radio"
          id="test_val2__input-radio__val2"
          name="test_val2"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="val2"
        />
        <label
          htmlFor="test_val2__input-radio__val2"
          onClick={[Function]}
        >
          <span>
            <img
              alt="Unilingual label"
              className="faIconLeft"
              src="img/test.png"
              style={
                Object {
                  "height": "3em",
                  "margin": "0 0.25em 0 0",
                }
              }
            />
          </span>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Unilingual label",
              }
            }
          />
        </label>
      </div>
      <input
        autoComplete="none"
        className="apptr__form-input input-custom apptr__input-string"
        id="foo.test.custom"
        name="foo.test.custom"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        tabIndex={-1}
        type="text"
        value=""
      />
    </div>
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_conditionalVal__input-radio__conditionalVal"
          name="test_conditionalVal"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="conditionalVal"
        />
        <label
          htmlFor="test_conditionalVal__input-radio__conditionalVal"
          onClick={[Function]}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "english conditional",
              }
            }
          />
        </label>
      </div>
    </div>
  </div>
  <div
    className="survey-question__input-radio-group-column"
  >
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_val3__input-radio__val3"
          name="test_val3"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="val3"
        />
        <label
          htmlFor="test_val3__input-radio__val3"
          onClick={[Function]}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Translated string",
              }
            }
          />
        </label>
      </div>
    </div>
    <div
      className="survey-question__input-radio-input-container"
      onClick={[Function]}
    >
      <div
        className="label-input-container"
      >
        <input
          checked={false}
          className="input-radio"
          id="test_custom__input-radio__custom"
          name="test_custom"
          onChange={[Function]}
          onClick={[Function]}
          type="radio"
          value="custom"
        />
        <label
          htmlFor="test_custom__input-radio__custom"
          onClick={[Function]}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "Other",
              }
            }
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;
